toolkit/tools/imagegen/installutils/installutils.go

```
@@ -1285,6 +1397,36 @@ func copyAdditionalFiles(installChroot *safechroot.Chroot, config configuration.
 	return
 }
 
+// cleanupRpmDatabase removes RPM database if the image does not require a package manager.
+// rootPrefix is prepended to the RPM database path - useful when RPM database resides in a chroot and cleanupRpmDatabase can't be called from within the chroot.
+// packagesToInstall is a list of packages that will be installed on the image
+func cleanupRpmDatabase(rootPrefix string, packagesToInstall []string) {
+	logger.Log.Info("Attempting RPM database cleanup...")
+	// If the image doesn't contain the package manager
+	// We can remove the RPM database files
+	rpmInChroot := false
+
+	for _, name := range []string{"rpm", "dnf", "tdnf", "yum"} {
+		if sliceutils.Find(packagesToInstall, name) != -1 {
+			logger.Log.Infof(`Package manager "%s" found in package list. Keeping the RPM database.`, name)
+			rpmInChroot = true
+			break
+		}
+	}
+
+	if !rpmInChroot {
+		logger.Log.Info("No package manager found in package list. Removing the RPM database.")
+		rpmDir := strings.Join([]string{rootPrefix, rpmDependenciesDirectory}, "")
+		err := os.RemoveAll(rpmDir)
+		if err != nil {
+			logger.Log.Errorf("Failed to remove RPM database (%s). Error: %s", rpmDir, err)
```

Hides error, should be returned and constitute a build break.